lib/rules/func-style.js

```diff
@@ -63,7 +81,18 @@ module.exports = {
             FunctionExpression(node) {
                 stack.push(false);
 
-                if (enforceDeclarations && node.parent.type === "VariableDeclarator") {
+                if (
+                    enforceDeclarations &&
+                    node.parent.type === "VariableDeclarator" &&
+                    typeof exportFunctionStyle === "undefined"
+                ) {
+                    context.report({ node: node.parent, messageId: "declaration" });
+                }
```

Comment:
Similar to the previous, this would be a false negative:

```js
/*eslint func-style: ["error", "declaration", { "overrides": { "namedExports": "expression" } }]*/

var foo = function() { // false negative
    // ...
};

```